### PR TITLE
Name claim's secrets by test

### PR DESCRIPTION
**Pull request by hongkailiu**
https://issues.redhat.com/browse/DPTP-2332

/cc @petr-muller

### CODE REVIEW

**Comment by hongkailiu** [comment]
/hold
expecting e2e to fail.

File: pkg/steps/cluster_claim.go [hunk 1]
```diff
@@ -175,7 +175,11 @@ func (s *clusterClaimStep) acquireCluster(ctx context.Context, waitForClaim func
 	return claim, nil
 }
 
-func getHiveSecret(src *corev1.Secret, name, namespace string) (*corev1.Secret, error) {
+func namePerTest(name, testName string) string {
+	return fmt.Sprintf("%s-%s", testName, name)
+}
```
Comment (petr-muller) [inline pkg/steps/cluster_claim.go:180]: I'm worried about the length - secret names have limited length. Job names are sometimes fairly long and our suffix is too. I think we can just prune the testname to be sure we can append a suffix with known length safely?

Comment (hongkailiu) [reply pkg/steps/cluster_claim.go:181]: Updated.
Also replace "." by "-".

**Review by petr-muller — APPROVED** [review]

**Pull request merged**